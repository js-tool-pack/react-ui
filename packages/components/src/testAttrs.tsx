import { fireEvent, render } from '@testing-library/react';
import type { PropsBase } from '@pkg/shared';
import React from 'react';

export function testAttrs<
  T extends { attrs?: Partial<React.HTMLAttributes<HTMLElement>> },
>(component: React.FC<T>, displayName?: string) {
  const name = displayName || component.displayName || component.name;
  test(name ? `${name} attrs` : `attrs`, () => {
    const onClick = jest.fn();

    const FC = component as React.FC<PropsBase>;
    const { container } = render(
      <FC
        attrs={{
          style: { width: '20px' },
          className: 'foo',
          role: 'note',
          onClick,
        }}
      />,
    );

    expect(container.firstChild).toHaveClass('foo');
    expect(container.firstChild).toHaveAttribute('role', 'note');
    expect(container.firstChild).toHaveStyle({ width: '20px' });

    expect(onClick).not.toBeCalled();
    fireEvent.click(container.firstChild!);
    expect(onClick).toBeCalled();
  });
}
