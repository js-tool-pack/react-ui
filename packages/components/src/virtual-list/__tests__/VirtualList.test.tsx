import { render } from '@testing-library/react';
import { testAttrs } from '~/testAttrs';
import { VirtualList } from '..';

describe('VirtualList', () => {
  testAttrs(VirtualList);
  it('basic', () => {
    expect(
      render(
        <VirtualList>
          <div>123</div>
          <div>123</div>
          <div>123</div>
        </VirtualList>,
      ).container.firstChild,
    ).toMatchSnapshot();
  });
});
