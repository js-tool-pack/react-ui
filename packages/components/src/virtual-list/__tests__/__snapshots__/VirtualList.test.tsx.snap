// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VirtualList basic 1`] = `
<div
  class="t-virtual-list"
>
  <div
    class="t-virtual-list__wrapper"
  >
    <div
      class="t-virtual-list__items"
      style="transform: translateY(0px);"
    >
      <div>
        123
      </div>
      <div>
        123
      </div>
      <div>
        123
      </div>
    </div>
  </div>
</div>
`;
